docs/upgrade/UPGRADE-v8.1.0-dev.md

```diff
@@ -125,7 +125,22 @@ There you can find links to upgrade notes for other versions too.
         + docker-compose exec -T php-fpm composer install
         + docker-compose exec -T php-fpm ./phing db-create test-db-create build-demo-dev-quick error-pages-generate
         ```
-
+- Run `php phing fix-annotations` to fix or add all the relevant annotations for your extended classes ([#1344](https://github.com/shopsys/shopsys/pull/1344))
+    - Thanks to the fixes, your IDE (PHPStorm) will understand your code better.
+    - You can read more about the whole topic in the ["Framework extensibility" article](../introduction/framework-extensibility.md#making-the-static-analysis-understand-the-extended-code).
+    - The checks and fixes of the proper annotations are now included in all `standards-*` phing targets, if you want to turn this feature off, add the following line into your `build.xml`:
+    ```diff
+  + <property name="check-and-fix-annotations" value="false"/>
+    ```
+- Increase your PHPStan level to 4 in your `build.xml` ([#1040](https://github.com/shopsys/shopsys/pull/1040))
+    ```diff
+  - <property name="phpstan.level" value="1"/>
+  + <property name="phpstan.level" value="4"/>
+    ```
+    - A lot of the possible issues should be already resolved if you followed the previous instruction and ran the `php phing fix-annotations` phing command.
+    - Some of the issues related to class extension need to be addressed manually nevertheless(see [the commit in monorepo](https://github.com/shopsys/shopsys/commit/efd008b8d))
+        - this is in such a case when you are working directly with an extended entity that you obtain as a result of a method call on not extended service (e.g. `FrameworkBundle\ProductFacade::getById()`)
+        - as a workaround, you can create an empty class extending the one from the framework, register the extension in your `services.yml`, and use `php phing fix-annotations` to generate appropriate annotations.
```

Comment:
In referenced commit are results extracted into a separated annotated variable. But from this it seems like creating empty class is the only and the right way how to solve the problem

Reply:
really? TBH I do not know what is the right way how to solve the problem, it really depends on the situation - if the developer is likely to extend such a class sooner or later, or it is reported in many places, it is better to create the empty extension, otherwise, I would go with extracting variables as it is quicker :slightly_smiling_face: Hm, I will write this into the docs :grin:

Reply:
I have added the information to docs in 80e3bc6dd and removed it from upgrade notes in 92461f66b to avoid information duplication. With so many fixups, I am not sure which information was added first, but I will pay attention to it during rebase (first, docs must be added, then upgrade information with the link to the docs)

Reply:
Nice! thanks